src/auth/auth_azure_ad_oauth.py

```diff
@@ -0,0 +1,34 @@
+import logging
+
+import tornado.auth
+
+from auth.auth_abstract_oauth import AbstractOauthAuthenticator, _OauthUserInfo
+from model import model_helper
+
+LOGGER = logging.getLogger('script_server.AzureADOauthAuthenticator')
+
+
+class AzureAdOAuthAuthenticator(AbstractOauthAuthenticator):
+    def __init__(self, params_dict):
+        params_dict['group_support'] = False
+        self.auth_url = model_helper.read_obligatory(params_dict, 'auth_url', ' for OAuth')
+        self.token_url = model_helper.read_obligatory(params_dict, 'token_url', ' for OAuth')
+
+        super().__init__(
+            self.auth_url,
+            self.token_url,
+            'openid email profile',
+            params_dict,
+        )
+
+    async def fetch_user_info(self, access_token) -> _OauthUserInfo:
+        headers = {'Authorization': f'Bearer {access_token}'}
+        user_response = await self.http_client.fetch('https://graph.microsoft.com/v1.0/me', headers=headers)
```

Comment:
Is the host always `'https://graph.microsoft.com/v1.0/me'` or it can be in a private namespace or on premise?

Reply:
Yes, my understanding is the URL is always this for Azure AD until v2.0 comes out. [When v2.0 comes out, the v1.0 endpoint will remain supported for 2 years following the announcement](https://learn.microsoft.com/en-us/graph/versioning-and-support).

Azure AD is strictly cloud-based, a private/on-prem setup would be using [Active Directory Domain Services](https://learn.microsoft.com/en-us/windows-server/identity/ad-ds/get-started/virtual-dc/active-directory-domain-services-overview) instead.

Reply:
thanks!